awswrangler/s3/_wait.py

```diff
@@ -1,33 +1,35 @@
 """Amazon S3 Wait Module (PRIVATE)."""
 
-import concurrent.futures
 import itertools
 import logging
-from typing import List, Optional, Tuple, Union
+from typing import List, Optional, Union
 
 import boto3
 
 from awswrangler import _utils
+from awswrangler._config import config
+from awswrangler._threading import _get_executor
+from awswrangler.distributed import ray_get, ray_remote
 
 _logger: logging.Logger = logging.getLogger(__name__)
 
 
 def _wait_object(
-    path: Tuple[str, str], waiter_name: str, delay: int, max_attempts: int, boto3_session: boto3.Session
+    boto3_session: Optional[boto3.Session], path: str, waiter_name: str, delay: int, max_attempts: int
 ) -> None:
     client_s3: boto3.client = _utils.client(service_name="s3", session=boto3_session)
     waiter = client_s3.get_waiter(waiter_name)
-    bucket, key = path
+
+    bucket, key = _utils.parse_path(path=path)
     waiter.wait(Bucket=bucket, Key=key, WaiterConfig={"Delay": delay, "MaxAttempts": max_attempts})
 
 
-def _wait_object_concurrent(
-    path: Tuple[str, str], waiter_name: str, delay: int, max_attempts: int, boto3_primitives: _utils.Boto3PrimitivesType
+@ray_remote
```

Comment:
I would like to specify here that the task is only allowed to take 25% of the cluster's available CPUs. This can be specified with the vanilla `ray.remote` function. However, we need to add support for it in our own `ray_remote` decorator. I've created an issue here: https://github.com/aws/aws-sdk-pandas/issues/1563